hdfs/client.py

```diff
@@ -150,38 +151,86 @@ class Client(object):
 
   def __init__(self, url, root=None, proxy=None, timeout=None, session=None):
     self.root = root
-    self.url = url
+    # We build a deque of all URLs
+    self._urls = deque([ u.rstrip('/') for u in url.split(';') if u ])
+    # Then we store what URL we'll start at when making requests
+    self._start_url_index = 0
     self._session = session or rq.Session()
     self._proxy = proxy
     self._timeout = timeout
     _logger.info('Instantiated %r.', self)
 
+  @property
+  def urls(self):
+    return list(self._urls)
+
   def __repr__(self):
-    return '<%s(url=%r)>' % (self.__class__.__name__, self.url)
+    return '<%s(url=%r)>' % (self.__class__.__name__, self._urls)
 
   # Generic request handler
-
   def _request(self, method, url, strict=True, **kwargs):
     """Send request to WebHDFS API.
 
     :param method: HTTP verb.
-    :param url: Url to send the request to.
+    :param url: Url to send the request to, or URL suffix to add to base.
     :param \*\*kwargs: Extra keyword arguments forwarded to the request
       handler. If any `params` are defined, these will take precendence over
       the instance's defaults.
 
     """
-    response = self._session.request(
-      method=method,
-      url=url,
-      timeout=self._timeout,
-      headers={'content-type': 'application/octet-stream'}, # For HttpFS.
-      **kwargs
-    )
-    if strict and not response: # Non 2XX status code.
-      return _on_error(response)
+    # If passed a full URL, just send the request
+    if url.startswith('http'):
```

Comment:
Funnily enough, I actually tried this implementation first.  But I guess I didn't fully understand the client yet because I couldn't get it working.  Done this way now.